meilisearch-release-changelog/v1.8.0.md

```diff
@@ -0,0 +1,219 @@
+}
+```
+
+Here is an example of setting OpenAI embedder with the `rest` source:
+
+```json5
+{
+  "source": "rest",
+  "apiKey": "<your-openai-api-key>",
+  "dimensions": 1536,
+  "url": "https://api.openai.com/v1/embeddings",
+  "query": {
+    "model": "text-embedding-ada-002"
+  },
+  "inputField": ["input"],
+  "inputType": "textArray",
+  "pathToEmbeddings": ["data"],
+  "embeddingObject": ["embedding"]
+}
+```
+
+**Ollama model**
+
+Here is how to set up the Ollama model:
+
+```json5
+"default": {
+  "source": "ollama", // 👈 Use the Ollama source
+  "url": "http://localhost:11434/api/embeddings",
+  // ☝️ optional, fetched from MEILI_OLLAMA_URL environment variable if missing
+  "apiKey": "<foobarbaz>",
+  // ☝️ optional
+  "model": "nomic-embed-text",
+  "documentTemplate": "blabla" // like for openAI and huggingFace sources
+}
+
+```
+
+### Expose the `distribution` shift setting
+
+When setting an embedder, you can now set the `distribution` shift.
+
+```json5
+"default": {
+  "source": "huggingFace", // supported for any source
+  "model": "some/model",
+  "distribution": {  // describes the natural distribution of results
+    "mean": 0.7, // mean value
+    "sigma": 0.3 // variance
+  }
+}
+```
+
+The “distribution shift” is an affine transformation applied to the `_rankingScore` of a **semantic** search result with the aim of making the comparison to the `_rankingScore` of a **keyword** search result more meaningful.
+
+### Other hybrid search improvements
+
+* Hide the API key in settings and task queue (#4533) @dureuill
+* Return the keyword search results even in case of a failure of the embedding (#4548) @dureuill
+* For hybrid or semantic search requests, add a `semanticHitCount` field at the top of the search response indicating the number of hits originating from the semantic search (#4548) @dureuill
+
+## Support negative keyword when searching
+
+Search queries can now contain a negative keyword to exclude terms from the search. Use the `-` operator in front of a word or a phrase to make sure no document that contains those words are shown in the results.
+
+- `-escape` returns a placeholder search without any document contains the `escape` word.
+- `-escape room` returns only documents containing the `room` word but not the `escape` one.
+- `-"on demand"` returns a placeholder search but without any document containing the `"on demand"` phrase.
+
+Done by @Kerollmops in #4535.
+
+## Search robustness improvements
+
+### Add a search cutoff
+
+To avoid any crash and performance issues, Meilisearch now stops search requests lasting more than 150ms.
+
+If you want to customize this value, you can update the `searchCutoffMs` settings (value in ms):
+
+```bash
+curl \
+  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "searchCutoffMs": 50
+  }'
+```
+
+The default value of the `searchCutoffMs` setting is `null` and corresponds to 150ms.
```

Comment:
````suggestion
### Search cut-off

To avoid crashes and performance issues, Meilisearch now interrupts search requests that take more than 150ms to complete.

Use the `/settings` endpoint to customize this value:

```bash
curl \
  -X PATCH 'http://localhost:7700/indexes/movies/settings' \
  -H 'Content-Type: application/json' \
  --data-binary '{
    "searchCutoffMs": 50
  }'
```

The default value of the `searchCutoffMs` setting is `null` and corresponds to a 150ms timeout.
````